catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/history/HistorianTest.java

```diff
@@ -0,0 +1,462 @@
+            throws SourceUnavailableException, IngestException, StorageException {
+        Metacard metacard = getMetacardUpdatePair().get(0);
+        DeleteRequest deleteRequest = new DeleteRequestImpl("deleteRequest");
+        DeleteResponse deleteResponse = new DeleteResponseImpl(deleteRequest,
+                new HashMap<>(),
+                Collections.singletonList(metacard));
+
+        historian.version(deleteResponse);
+        assertThat(storageProvider.storageMap.size(), equalTo(0));
+    }
+
+    @Test
+    public void testNullContentItemStorageReadRequest()
+            throws StorageException, UnsupportedQueryException, SourceUnavailableException,
+            IngestException, URISyntaxException {
+        List<Metacard> metacards = getMetacardUpdatePair();
+
+        // Parameters for historian
+        UpdateStorageRequest storageRequest = mock(UpdateStorageRequest.class);
+        UpdateStorageResponse storageResponse = mock(UpdateStorageResponse.class);
+        UpdateResponse updateResponse = mock(UpdateResponse.class);
+
+        // Make content item null
+        storageProvider.storageMap.put(METACARD_ID, null);
+
+        mockQuery(metacards.get(1));
+        historian.version(storageRequest, storageResponse, updateResponse);
+
+        // Verify that the content item DIDN't updated
+        ReadStorageRequest request = mock(ReadStorageRequest.class);
+        when(request.getResourceUri()).thenReturn(new URI(RESOURCE_URI));
+        ContentItem item = storageProvider.read(request)
+                .getContentItem();
+
+        assertThat(item, nullValue());
+    }
+
+    @Test
+    public void testReadStorageException()
+            throws StorageException, SourceUnavailableException, IngestException,
+            UnsupportedQueryException, URISyntaxException {
+        StorageProvider exceptionStorageProvider = mock(StorageProvider.class);
+        when(exceptionStorageProvider.read(any())).thenThrow(StorageException.class);
+        historian.setStorageProviders(Collections.singletonList(exceptionStorageProvider));
+
+        Metacard metacard = getMetacardUpdatePair().get(0);
+
+        // Parameters for historian
+        UpdateStorageRequest storageRequest = mock(UpdateStorageRequest.class);
+        UpdateStorageResponse storageResponse = mock(UpdateStorageResponse.class);
+        UpdateResponse updateResponse = mock(UpdateResponse.class);
+
+        storeMetacard(metacard);
+
+        // send a request to update the metacard
+        updateMetacard(storageRequest, storageResponse, metacard);
+        storageProvider.update(storageRequest);
+
+        mockQuery(metacard);
+        historian.version(storageRequest, storageResponse, updateResponse);
+
+        // Verify that it wasn't updated
+        Metacard update = readMetacard();
+
+        assertThat(update, equalTo(metacard));
+    }
+
+    @Test(expected = IngestException.class)
+    public void testTryCommitStorageException()
```

Comment:
We basically need to make sure the `StorageException` is caught and ignored when `rollback` fails, but that a `IngestException` still gets thrown. The flow and expectation would be very similar to this test but with one more step, i.e., `commit` fails then `rollback` also fails.